src/scripts/context.py

```
@@ -0,0 +1,650 @@
+        if path.is_dir():
+            return {"path": str(path), "sha": meta["sha"],
+                    "fetched_at": meta.get("fetched_at"), "source": "cache"}
+    return None
+
+
+def sparse_fetch(project: str, record: dict):
+    remote, branch = record["remote"], record.get("branch", "main")
+    context_root = record.get("context_root", DEFAULT_KNOWLEDGE)
+    with tempfile.TemporaryDirectory() as tmp:
+        clone = Path(tmp) / "clone"
+        proc = subprocess.run(
```

Ruff (0.16.0)

[error] `subprocess` call: check for execution of untrusted input

(S603)